### PR TITLE
Branch remove edit constraint

### LINKED ISSUE #245: Remove edit past reservation restriction

### PR DESCRIPTION
**Pull request by Kaidama97**
Previously, users were not allowed to edit past reservations. This restriction has now been relaxed to allow users to modify reservation details, except for the date-time.

The original constraint prevented users from making any changes to past reservations, which was too restrictive.

Changes Implemented:
- Users can now edit details of past reservations (e.g., number of diners, customer information).
- Users can modify the date-time of past reservations.
- Future reservations cannot be changed to past dates.
- A warning message is displayed when editing past reservations.

This is the original data.
image

`edit 1 x/2 n/John Doe` - successfully edit, warning message shown.
image

`edit 1 d/2025-05-05 1000 `- edit past reservation future date success.
image

`edit 2 d/2025-05-05 1000` - success, edit future reservation.
image

`edit 2 d/2023-04-23 1900` - invalid dates fail past date
image

Fix #245

### CODE REVIEW

**Review by JohannsenLum — APPROVED** [review]
LGTM, just rmb to test all the edge cases incase

File: src/main/java/seedu/reserve/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -178,6 +178,21 @@ public static DateTime parseDateTime(String dateTime) throws ParseException {
         return new DateTime(trimmedDateTime);
     }
 
+    /**
+     * Parses a {@code String dateTime} into a {@code DateTime}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code dateTime} is invalid.
+     */
+    public static DateTime parseEditedDateTime(String dateTime) throws ParseException {
```
Comment (wowwwp) [inline src/main/java/seedu/reserve/logic/parser/ParserUtil.java:187]: Looks very similar to the filter prase date time, is it possible to merge?

File: src/main/java/seedu/reserve/model/reservation/DateTime.java [hunk 2]
```diff
@@ -62,6 +62,18 @@ public static DateTime fromFileString(String dateTime) {
         return new DateTime(LocalDateTime.parse(dateTime, FORMATTER));
     }
 
+    /**
+     * Constructs a {@code DateTime} from a string input without validation.
+     *
+     * @param dateTime A date-time string.
+     * @return A DateTime object if the string is valid, otherwise null.
+     */
+    public static DateTime fromEditedDateString(String dateTime) {
```
Comment (wowwwp) [inline src/main/java/seedu/reserve/model/reservation/DateTime.java:71]: Is it possible to use the `fromFileString(...)` function i believe the logic is similar

File: src/main/java/seedu/reserve/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -58,8 +58,9 @@ public class EditCommand extends Command {
 
     public static final String MESSAGE_EDIT_RESERVATION_SUCCESS = "Edited Reservation:\n%1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_FUTURE_RESERVATION_REQUIRED = "Past reservation cannot be edited.";
-
+    public static final String MESSAGE_EDIT_PAST_RESERVATION_WARNING = "Warning: You modified a past reservation!\n";
```
Comment (wowwwp) [inline src/main/java/seedu/reserve/logic/commands/EditCommand.java:61]: Is it better to omit the word past because it kind of gives the impression that it is past wrt to current time?

**Review by wowwwp — APPROVED** [review]
just a few cosmetic issues, lgtm!

File: src/main/java/seedu/reserve/logic/parser/ParserUtil.java (hunk 1 above)
Comment (Kaidama97) [reply src/main/java/seedu/reserve/logic/parser/ParserUtil.java:187]: This function constraints are slightly more relaxed. It doesn't check if the date time is before current time.
